scripts/benchmark_stats.py

```diff
@@ -1,77 +1,84 @@
+from collections import defaultdict
 from csv import DictReader
 from glob import glob
-from typing import List
+from statistics import mean
+from typing import List, Mapping
 
 
 class BenchmarkStats:
+    PERCENTILES_TO_REPORT = (50, 90, 95, 99, 99.9, 100)
+    PERCENTILES_TO_GRAPH = (50, 90, 95, 99)
+    PERCENTILE_TO_USE_FOR_ROUTES = 99
+
     def __init__(self, folder_paths: List[str]):
         self._files: List = []
         for folder_path in folder_paths:
             self._files.extend(glob(f"{folder_path}/*stats.csv"))
 
-        self.weighted_get_requests: List[int] = []
-        self.weighted_post_requests: List[int] = []
-        self._total_requests: int = 0
-        self._total_failures: int = 0
-        self._total_get_requests: int = 0
-        self._total_post_requests: int = 0
+        self._get_requests: List[int] = []
+        self._post_requests: List[int] = []
+
         self._process_file_data()
 
     def __str__(self):
+        formatted_percentiles = "\n".join(
+            f"{percentile}th: {self._percentiles[percentile]}ms"
+            for percentile in self.PERCENTILES_TO_REPORT
+        )
         return (
-            f'GETs average: {int(self.average_weighted_get)}ms\n'
-            f'POSTs average: {int(self.average_weighted_post)}ms\n'
-            f'All requests average: {int(self.average_weighted_total)}ms\n'
-            f'Total Requests: {int(self._total_requests):,}\n'
-            f'Total Failures: {int(self._total_failures):,}\n'
+            f'---\n'
+            f'Percentiles:\n'
+            f'{formatted_percentiles}\n'
+            f'---\n'
+            f'GETs average: {self.average_get}ms\n'
```

Comment:
Probably still needs the Heading 99%. Although this still looks weird on the results I am using, as both GET and POST are lower then in the percentiles block
```
---
Percentiles:
50th: 60ms
90th: 85ms
95th: 98ms
99th: 140ms
99.9th: 240ms
100th: 850ms
---
GETs average: 108ms
POSTs average: 131ms
---
Total Requests: 204,701
Total Failures: 0
Error Percentage: 0%
```

Reply:
As discussed, I think it's still worth outputting POST/GET statistics separately (particularly now we've removed that split from the graph) as although historically they have tracked each other, it might highlight future issues. Adding the fact they're 99th and clarifying somewhere in the code/readme why they wont straddle the 99th in the above table is probably good enough

Reply:
Have updated the output and readme. Take a look and let me know if it is sufficient.